### PR TITLE
started drag and drop implementation.

### PR DESCRIPTION
**Pull request by amad1101**
This is the error I am currently getting.
Screenshot at Apr 23 19-03-23

### CODE REVIEW

**Comment by Korede-TA** [comment]
use the code here as a guide: https://github.com/Korede-TA/integrated-spreadsheet-environment/blob/master/src/model.rs#L703-L706

**Comment by amad1101** [comment]
I fixed the expected error, however, this is cause a lifetime error which is shown as follow
Screenshot at Apr 24 13-11-28

**Comment by Korede-TA** [comment]
this is a tricky one. it’s hard to explain how to fix it but i could pull the PR myself and he’ll debug

**Comment by Korede-TA** [comment]
hey @amad1101 is this working now? please post screenshot and remember to notify me when you do.

**Comment by amad1101** [comment]
I had to look into the Recreate action and I the grid creation is working. The issue it causing right now is that the row below get deleted whenever I try to drop a CSV file and the grid get created. 

Del_Bel

**Comment by Korede-TA** [comment]
hmm, that's a weird bug. but i suspect it might be related to the bug with the AddNestedGrid button where it erases some cells. we'll have to seriously investigate that bug soon. cc: @SirD-AL @htl43 

in the meantime, keep working on getting the actual CSV data loaded into the sheet. also start to think about a testing strategy for this, we'll work on implementing them together during the dev meeting.

**Comment by amad1101** [comment]
Nested Grid creation on a Parent grammar with 1 Row and more than 1 Col Delete all rows below and reshape the given row it was created on.
@Korede-TA  Please look at nested grid Action!
Nested_1_3

**Comment by amad1101** [comment]
As a current status, I am able to insert values at the code level but I am getting an unexpected unwrap error that need some investigation. I suspect it's coming from the csv library. refer to picture below.
Creat_err

File: src/model.rs [hunk 1]
```diff
@@ -552,37 +552,60 @@ impl Component for Model {
...
+                        let row_ = coord_.0.get() as usize;
+                        let col_ = coord_.1.get() as usize;
+                        let mut fragments: Vec<(NonZeroU32, NonZeroU32)> = Vec::new();
+                        fragments.push(coord_.clone());
+                        let c = Coordinate {
+                                row_cols: fragments,
+                                };
+                        let grid_: &str = &grid[row_ - 1][col_ - 1];
+                        grammar.remove(&c);
+                        grammar.insert(c, Grammar::input("", grid_));
```
Comment (Korede-TA) [inline src/model.rs:603]: i'm not totally sure what you're doing here, but consider using `Coordinate::child_of` function

**Comment by amad1101** [comment]
Finished drag and drop implementation for CSV file type. Please make a last review.
finishe_drag_drop_imp

**Comment by htl43** [comment]
Cool. Nice done 👍
    	
    	

    	Get Outlook for iOS
    
  




On Sat, May 2, 2020 at 10:27 AM -0400, "Amadou Diallo" <notifications@github.com> wrote:















Finished drag and drop implementation for CSV file type. Please make a last review.

**Comment by Korede-TA** [comment]
looks great!! let's get this merged

**Pull request merged**